src/lines_to_params.rs

```diff
@@ -0,0 +1,165 @@
+                kwargs.set_item("value", value).unwrap();
+                kwargs.set_item("multiplier", multiplier).unwrap();
+                kwargs.set_item("additional_params", (param0, param1)).unwrap();
+                action_effect.call((), Some(kwargs)).unwrap()
+            }).collect::<Vec<&PyAny>>();
+
+        let kwargs = PyDict::new(py);
+        kwargs.set_item("timestamp", timestamp).unwrap();
+        kwargs.set_item("action_effects", action_effects).unwrap();
+        kwargs.set_item("source_actor", source_actor_obj).unwrap();
+        kwargs.set_item("target_actor", target_actor_obj).unwrap();
+        kwargs.set_item("ability", ability_event).unwrap();
+        kwargs.set_item("sequence_id", sequence).unwrap();
+
+        PyModule::import(py, "nari.types.event.ability").unwrap()
+            .getattr("Ability").unwrap().call((), Some(kwargs)).unwrap().to_object(py)
+    })
+}
+
+/// Params to action_effect
+#[pyfunction]
+#[pyo3(text_signature = "(params: list[str]) -> list[int]")]
+pub(crate) fn action_effect_from_params(inp: Vec<&str>) -> ActionEffectParams {
+    let mut num = parser::u32_from_param(inp[0]);
+    let param0 = (&num >> 24) as u8;
+    let param1 = (&num >> 16) as u8;
+    let severity = (&num >> 8) as u8;
+    let effect_category = num as u8;
+    num = parser::u32_from_param(inp[1]);
+    let value = (&num >> 16) as u16;
+    let flags = (&num >> 8) as u8;
+    let multiplier = num as u8;
+    (param0, param1, severity, effect_category, value, flags, multiplier)
+}
+
+/// Params to status_effect
+#[pyfunction]
+#[pyo3(text_signature = "(params: list[str]) -> (int, int, float, int")]
+pub(crate) fn status_effect_from_params(inp: Vec<&str>) -> StatusEffectParams {
+    let (param0, param1) = parser::u16x2_from_param(inp.get(0).unwrap());
+    (
+        param0,
+        param1,
+        parser::f32_from_param(inp.get(1).unwrap()),
+        parser::u32_from_param(inp.get(2).unwrap()),
+    )
+}
+
+/// Params to statuslist
+#[pyfunction]
+#[pyo3(text_signature = "(params: list[str]) -> list[any]")]
+pub(crate) fn statuslist_from_params(inp: Vec<&str>) -> StatuslistParams {
+    let mut col = inp;
+    let actor = col.drain(..2).collect::<Vec<&str>>();
+    let class = col.drain(..1).collect::<Vec<&str>>();
+    let resources = col.drain(..6).collect::<Vec<&str>>();
+    let position = col.drain(..4).collect::<Vec<&str>>();
+    let status_effects = col.drain(..col.len() - 1).collect::<Vec<&str>>();
+    (
+        parse_id_name_pair(actor),
+        class.first().unwrap(),
+        resources.iter().map(|x| parser::u32_from_param(x)).collect(),
+        position.iter().map(|x| parser::f32_from_param(x)).collect(),
+        status_effects
+            .chunks(3)
+            .map(|x| status_effect_from_params(x.to_vec()))
+            .collect(),
+    )
+}
+
+fn create_id_name_pair<'a>(py: Python<'a>, module: &'a str, class: &'a str, id_name_pair: Vec<&'a str>) -> &'a PyAny {
```

Comment:
Er.....

Reply:
Rust just being rust on typing and just more of a generic method to create types that are an `IdNamePair` in python for usage in rust when constructing the final object.

Reply:
Added a comment above the function definition line explaining what it does